core/src/main/java/org/geysermc/geyser/entity/type/living/ArmorStandEntity.java

```diff
@@ -101,22 +106,21 @@ public boolean despawnEntity() {
 
     @Override
     public void moveRelative(double relX, double relY, double relZ, float yaw, float pitch, float headYaw, boolean isOnGround) {
+        Vector3f newPosition = position.add(relX, relY, relZ);
         if (secondEntity != null) {
-            secondEntity.moveRelative(relX, relY, relZ, yaw, pitch, headYaw, isOnGround);
+            secondEntity.moveAbsolute(newPosition, yaw, pitch, headYaw, onGround, false);
         }
-        super.moveRelative(relX, relY, relZ, yaw, pitch, headYaw, isOnGround);
+        moveAbsolute(newPosition, yaw, pitch, headYaw, onGround, false);
     }
 
     @Override
     public void moveAbsolute(Vector3f position, float yaw, float pitch, float headYaw, boolean isOnGround, boolean teleported) {
         if (secondEntity != null) {
-            secondEntity.moveAbsolute(applyOffsetToPosition(position), yaw, pitch, headYaw, isOnGround, teleported);
-        } else if (positionRequiresOffset) {
-            // Fake the height to be above where it is so the nametag appears in the right location for invisible non-marker armour stands
-            position = applyOffsetToPosition(position);
+            secondEntity.moveAbsolute(position, yaw, pitch, headYaw, isOnGround, teleported);
         }
-
-        super.moveAbsolute(position, yaw, yaw, yaw, isOnGround, teleported);
+        // Fake the height to be above where it is so the nametag appears in the right location
+        super.moveAbsolute(position.up(getYOffset()), yaw, yaw, yaw, isOnGround, teleported);
```

Comment:
This is definitely cleaner, but due to the nature of Cloudburst Vectors at this time, this will create a new Vector object. It might be better to do a check for `!= 0`.